File: java-workshop/part-1-powermock/src/main/java/service/PaymentService.java
package service;

import domain.PaymentResult;
import domain.PaymentStatus;
import infrastructure.PaymentGateway;

/**
 * Payment service with hardcoded PaymentGateway dependency (static calls).
 */
public class PaymentService {

    public PaymentResult processPayment(String orderId, String customerId, double amount) {
        if (amount <= 0) {
            return new PaymentResult(orderId, PaymentStatus.FAILED, null);
        }

        PaymentResult gatewayResult = PaymentGateway.processPayment(customerId, amount);
        return new PaymentResult(orderId, gatewayResult.status(), gatewayResult.transactionId());
    }

    public boolean refundPayment(String transactionId) {
        return PaymentGateway.refund(transactionId);
    }
}

File: csharp-workshop/Part1.StaticDeps/Service/PaymentService.cs
using Part1.StaticDeps.Domain;
using Part1.StaticDeps.Infrastructure;

namespace Part1.StaticDeps.Service;

/// <summary>
/// Payment service with hardcoded PaymentGateway dependency (static calls).
/// </summary>
public class PaymentService
{
    public PaymentResult ProcessPayment(string orderId, string customerId, double amount)
    {
        if (amount <= 0)
        {
            return new PaymentResult(orderId, PaymentStatus.Failed, null);
        }

        var gatewayResult = PaymentGateway.ProcessPayment(customerId, amount);
        return new PaymentResult(orderId, gatewayResult.Status, gatewayResult.TransactionId);
    }

    public bool RefundPayment(string transactionId) =>
        PaymentGateway.Refund(transactionId);
}
